### PR TITLE
fix: Don't build uwsgi with xml support.

### PR DESCRIPTION
**Pull request by feanil**
When uwsgi runs with xml support, it throws the following error on startup with the latest version of edx-platform:

```
xmlsec.InternalError: (-1, 'lxml & xmlsec libxml2 library version mismatch')
```

See https://github.com/xmlsec/python-xmlsec/issues/320 for more details on this.

Essentially, the uWSGI wheel is built against a different version of libxml2 and it causes this version mismach error when trying to load xmlsec.  The xml support in uWSGI is only needed for runing with xml configuration files, which tutor does not do.  Following the guidance of the above issue, I updated the `openedx` Dockerfile to no longer compile with any xml support as a part of the build. The time to build uWSGI was only slightly more than building from cache so I'm not concerned about major slowdowns in the build time for un-cached builds.

### CODE REVIEW

**Review by regisb — COMMENTED** [review]
This seems to be one of the first ripple effects of the pending [uwsgi deprecation](https://github.com/overhangio/tutor/issues/937)... :fearful: 

Since this is fixing a bug, can you please add a changelog entry? https://docs.tutor.edly.io/tutor.html#contributing

**Comment by DawoudSheraz** [comment]
Since ubuntu PR was merged in nightly, this PR should be merged to nightly as well. Apart from changelog, you might want to rebase the branch on nightly.

**Comment by feanil** [comment]
Added a changelog entry, let me know if you need anything else.

File: changelog.d/20241113_131448_feanil_fix_uwsgi_xml_issues_master.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+- Indicate breaking changes by prepending an explosion 💥 character.
+- Prefix your changes with either [Bugfix], [Improvement], [Feature], [Security], [Deprecation].
+- You may optionally append "(by @<author>)" at the end of the line, where "<author>" is either one (just one)
+  of your GitHub username, real name or affiliated organization. These affiliations will be displayed in
+  the release notes for every release.
+-->
+
+<!-- - 💥[Feature] Foobarize the blorginator. This breaks plugins by renaming the `FOO_DO` filter to `BAR_DO`. (by @regisb) -->
+<!-- - [Improvement] This is a non-breaking change. Life is good. (by @billgates) -->
+
```
Comment (DawoudSheraz) [inline changelog.d/20241113_131448_feanil_fix_uwsgi_xml_issues_master.md:12]: please remove this default text from changelog.

File: changelog.d/20241113_131448_feanil_fix_uwsgi_xml_issues_master.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+[Bugfix] Don't build uwsgi with XML support (by @feanil)
```
Comment (DawoudSheraz) [inline changelog.d/20241113_131448_feanil_fix_uwsgi_xml_issues_master.md:13]:
```suggestion
- [Bugfix] Don't build uwsgi with XML support (by @feanil)
```

**Review by DawoudSheraz — APPROVED** [review]

**Comment by DawoudSheraz** [comment]
@regisb Can you take another look at this, please? Thanks

**Review by regisb — APPROVED** [review]
lgtm! we should now split the django-redis and uwsgi installations, because they use different installation instructions, but we can do this in a separate PR.

**Pull request merged**